packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -0,0 +1,129 @@
+  const sourceFsPath = sourceUri.fsPath;
+  const inputOptions = {
+    placeHolder: nls.localize(RENAME_INPUT_PLACEHOLDER),
+    promopt: nls.localize(REAME_INPUT_PROMPT)
+  } as vscode.InputBoxOptions;
+
+  const responseText = await vscode.window.showInputBox(inputOptions);
+  if (sourceFsPath) {
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new FilePathGatherer(sourceUri),
+      new RenameLwcComponentExecutor(sourceFsPath, responseText)
+    );
+    await commandlet.run();
+  }
+}
+
+function renameComponent(sourceFsPath: string, newName: string) {
+  const componentPath = getComponentPath(sourceFsPath);
+  const componentName = path.basename(componentPath);
+  checkForDuplicateName(componentPath, newName);
+  const items = fs.readdirSync(componentPath);
+  if (items) {
+    for (const item of items) {
+      // only rename the file that has same name with component
+      if (isNameMatch(item, componentName, componentPath)) {
+        const newItem = item.replace(componentName, newName);
+        fs.renameSync(
+          path.join(componentPath, item),
+          path.join(componentPath, newItem));
+      }
+    }
+  }
+  const newComponentPath = path.join(path.dirname(componentPath), newName);
+  fs.renameSync(
+    componentPath,
+    newComponentPath);
+  notificationService.showWarningMessage(nls.localize(RENAME_COMP_WARNING));
+}
+
+function getComponentPath(sourceFsPath: string): string {
+  const stats = fs.statSync(sourceFsPath);
+  return stats.isFile() ? path.dirname(sourceFsPath) : sourceFsPath;
+}
+
+function checkForDuplicateName(componentPath: string, newName: string) {
+  if (isDuplicate(componentPath, newName)) {
+    notificationService.showErrorMessage(nls.localize(REANME_INPUT_DUP_ERROR));
+    throw new Error(format(nls.localize(REANME_INPUT_DUP_ERROR)));
+  }
+}
+
+function isDuplicate(componentPath: string, newName: string): boolean {
+  const isLwc = isLwcComp(componentPath);
+  const lwcPath = isLwc ? path.dirname(componentPath) : path.join(path.dirname(path.dirname(componentPath)), 'lwc');
+  const auraPath = isLwc ? path.join(path.dirname(path.dirname(componentPath)), 'aura') : path.dirname(componentPath);
+  if (fs.existsSync(path.join(lwcPath, newName)) || fs.existsSync(path.join(auraPath, newName))) {
+    return true;
+  }
+  return false;
+}
+
+function isNameMatch(item: string, componentName: string, componentPath: string) {
+  const isLwc = isLwcComp(componentPath);
+  let regularExp: RegExp;
+  if (isLwc) {
+    regularExp = new RegExp(`${componentName}\.(html|js|js-meta.xml|css|svg)`);
+  } else {
+    regularExp = new RegExp(`${componentName}(((Controller|Renderer|Helper)?\.js)|(\.(cmp|app|css|design|auradoc|svg)))`);
+  }
+  return item.match(regularExp) ? true : false;
+}
+
+function isLwcComp(componentPath: string): boolean {
+  return path.basename(path.dirname(componentPath)) === 'lwc' ? true : false;
```

Comment:
since a string comparison is taking place, a call to `.toLowerCase()` should probably be added.
```
return path.basename(path.dirname(componentPath)).toLowerCase() === 'lwc' ? true : false;
```

Reply:
Is the function `isLwcComp` case sensitive?

Reply:
@RitamAgrawal @jeffb-sfdc  'lwc' folder in a SFDX project is always lower-case, which is defined when user create a new SFDX project.

Reply:
Got it.